database-mongo/index.js

```diff
@@ -13,22 +13,83 @@ db.once('open', function() {
 });
 
 var games = mongoose.Schema({
-  token: String,
-  players: String,
-  results: String  
+  token: {type: String, required: true},	
+  playerRoles: String,
+  results: String, 
+  playerIds: String,
 });
 
-var Game = mongoose.model('Game', games);
+module.exports.Game = mongoose.model('Game', games);
 
-var selectGame = function(token, callback) {
-  Game.find({token});
+module.exports.selectGame = function(token, callback) {
+  Game.find({token}, (err, game) => {
+  	console.log(game);
+  	callback(game);
+  });
 };
 
-var createGame = function(token, players, results, callback) {
+module.exports.createGame = function(token, hostName, clientId, callback) {
+	Game.create({token}, (err) => {
+		if (err) {
+			callback(err);
+		} else {
+			var playerRoles = {};
+			playerIds[hostName] = clientId;
+			var playerIds = JSON.stringify(playerIds);
+			var results = [];
+			Game.update({token}, {playerIds, results}, (err) => {
+				callback(err);
+			});
+		}
+	});
 };
 
-var updateResults = function (results, callback) {
+module.exports.addPlayer = function(token, playerName, clientId, callback) {
+	Game.find({token}, (err, game) => {
+		var playerIds = JSON.parse(game.playerIds);
+		playerIds[playerName] = clientId;
+		playerIds = JSON.stringify(playerIds);
+		Game.update({token}, {playerIds}, (err) => {
+			callback(err);
+		};
+	});
 };
 
+module.exports.removePlayer = function(token, playerName, callback) {
+	Game.find({token}, (err, game) => {
+		var playerIds = JSON.parse(game.playerIds);
+		delete playerIds[playerName];
+		playerIds = JSON.stringify(playerIds);
+		Game.update({token}, {playerIds}, (err) => {
+			callback(err);
+		};
+	});
+};
+
```

Comment:
Is the intent to use .removePlayer to handle when a player opts to leave a game while waiting for players to join?